src/compiler/checker.ts

```diff
@@ -4499,7 +4525,19 @@ namespace ts {
                         if (source === numberType && target.flags & TypeFlags.Enum) return Ternary.True;
                     }
                 }
+
+                if (relation === assignableRelation && source.flags & TypeFlags.ObjectLiteral && source.flags & TypeFlags.FreshObjectLiteral) {

```

Comment:
Yeah, `TypeFlags.FreshObjectLiteral` already implies both, so it's enough to just check for that.

Agreed, we should do the check for subtype as well.